llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -1969,6 +1973,11 @@ void LazyValueInfo::threadEdge(BasicBlock *PredBB, BasicBlock *OldSucc,
   }
 }
 
+void LazyValueInfo::eraseValue(Value *V, const Module *M) {
+  if (PImpl)
+    getImpl(PImpl, AC, M).eraseValue(V);
```

Comment:
This is a bit awkward, because we only really need getImpl() if PImpl is not initialed yet, which it already is here. This results in the unnecessary Module argument. Can we just directly do the static_cast here? Or add a getImpl() variant that only returns LazyValueInfoImpl* if it is already initialized (and null otherwise)?

Reply:
I'll try to refactor it.

Reply:
The `M` makes no sense. I just deleted it.
I want to backport the first two commits. So I will split the refactoring into a third commit.

Reply:
Yeah, splitting up the refactoring is a good idea.